Use layout manager for Create Wallet dialog

hebasto opened the pull request:
On master (e75f91eae3936269b40b4bfdfe540d5526270936) not using layout manager causes problems with resizing:

![Screenshot from 2021-01-01 13-03-13](https://user-images.githubusercontent.com/32963518/103437728-ce1d4580-4c33-11eb-8915-1e9482775653.png)
![Screenshot from 2021-01-01 13-03-26](https://user-images.githubusercontent.com/32963518/103437730-d6758080-4c33-11eb-9e0f-87d0dd487fcb.png)

Also text labels are not resized properly on some window managers (https://github.com/bitcoin/bitcoin/issues/20777), or if their lengths are changed (after translation).

This PR introduces a standard layout manager for the "Create Wallet" dialog that fixes all layout issues (actually, the `createwalletdialog.ui` has been re-written from scratch):

![Screenshot from 2021-01-01 13-10-03](https://user-images.githubusercontent.com/32963518/103437822-d0cc6a80-4c34-11eb-84fd-fcb10a16d9ef.png)
![Screenshot from 2021-01-06 23-50-36](https://user-images.githubusercontent.com/32963518/103823090-0b416780-507a-11eb-89dd-3f48a358e168.png)

Additional visual changes:
- advanced options are grouped in `QGroupBox` (https://github.com/bitcoin-core/gui/pull/96#issuecomment-726337165)
- enabled the [size grip](https://doc.qt.io/qt-5/qsizegrip.html#details)

Fix https://github.com/bitcoin/bitcoin/issues/20777

Sjors commented:
Concept ACK

dooglus commented:
Looks great now:

![Screenshot_2021-01-01_14-05-35](https://user-images.githubusercontent.com/573356/103447010-851cbf80-4c3a-11eb-94eb-0eb8c05f5e94.png)

When I resize the window it redraws like this:

![Screenshot_2021-01-01_14-05-52](https://user-images.githubusercontent.com/573356/103447015-906feb00-4c3a-11eb-818f-c00afb4e2777.png)

MarcoFalke commented:
Tagged for 0.21.1

jarolrod reviewed (commented):
Testing on macOS Big Sur 11.1 with Qt 5.15.2 I noticed some abnormalities concerning the resizing behavior and the text. Below are some screenshots comparing behavior of `Master` and `PR`. Looking at the `PR` screenshots you will notice:
1) The `Wallet Input Form` does not stay in the top left corner and it does not resize. It just stays in the center of the window. 
2) The `Advance Options` Text is no longer Bold
3) The `Advanced Options` and `Wallet Name` text are a different size no longer aligned


**Master**
<img width="374" alt="master-normal" src="https://user-images.githubusercontent.com/23396902/103726909-001e0700-4fa8-11eb-92dd-6aee9b2ec7ea.png">


<img width="852" alt="master-resize" src="https://user-images.githubusercontent.com/23396902/103726914-0318f780-4fa8-11eb-9513-ae8ccb4e641d.png">


**PR**
<img width="394" alt="pr-normal" src="https://user-images.githubusercontent.com/23396902/103726994-2d6ab500-4fa8-11eb-94c2-da803234c96b.png">

<img width="764" alt="pr-resize" src="https://user-images.githubusercontent.com/23396902/103727002-2f347880-4fa8-11eb-8c48-5a3d42547fde.png">

jonatack commented:
Concept ACK

luke-jr commented:
I would think the empty space belongs below the advanced-options group box...

hebasto commented:
Updated 3cbcb251c24afc707fe53c360dd7b3b3fd7055f0 -> d4feb6812a2707ef85d75dda4372086ec62eb922 ([pr171.01](https://github.com/hebasto/gui/commits/pr171.01) -> [pr171.02](https://github.com/hebasto/gui/commits/pr171.02), [diff](https://github.com/hebasto/gui/compare/pr171.01..pr171.02)):

- addressed @jarolrod's [comment](https://github.com/bitcoin-core/gui/pull/171#pullrequestreview-562308613) for macOS:
> The `Wallet Input Form` does not stay in the top left corner and it does not resize. It just stays in the center of the window.

- addressed @luke-jr's [comment](https://github.com/bitcoin-core/gui/pull/171#issuecomment-755455321):
> I would think the empty space belongs below the advanced-options group box...

---

@jarolrod 
> 2. The `Advance Options` Text is no longer Bold
> 
> 3. The `Advanced Options` and `Wallet Name` text are a different size

It is intended to left default parameters for the `QGroupBox` widget in order to make look-and-feel standard for a particular platform. We already use the same approach, see screenshots from macOS:
![Screenshot from 2021-01-06 23-48-20](https://user-images.githubusercontent.com/32963518/103822949-c1f11800-5079-11eb-8fdb-559467bfa010.png)
![Screenshot from 2021-01-06 23-47-48](https://user-images.githubusercontent.com/32963518/103822953-c4ec0880-5079-11eb-97f2-ee46f814575b.png)

---

OP screenshots updated.

jarolrod commented:
d4feb6812a2707ef85d75dda4372086ec62eb922 Fixed the resizing issue on macOS for me. 
<img width="788" alt="Screen Shot 2021-01-06 at 5 16 04 PM" src="https://user-images.githubusercontent.com/23396902/103825429-08775000-5043-11eb-8d6f-89c9531a6c5e.png">

Sjors commented:
re-tACK d4feb6812a2707ef85d75dda4372086ec62eb922

promag commented:
Why not prevent vertical resize? Is there a guideline to support unnecessary vertical resize?

hebasto commented:
Yes, vertical resize is unnecessary.
But why restrict users?

On Fri, 8 Jan 2021 at 19:00 João Barbosa <notifications@github.com> wrote:

> Why not prevent vertical resize? Is there a guideline to support
> unnecessary vertical resize?
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin-core/gui/pull/171#issuecomment-756873020>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH3PXPXAOBV4GK6ZMVVZQKDSY422TANCNFSM4VQGTD6A>
> .
>
-- 

--
Hennadii Stepanov

laanwj commented:
Concept ACK

promag reviewed (commented):
Tested ACK d4feb6812a2707ef85d75dda4372086ec62eb922 on macos.
